sae_lens/sae_training_runner.py

```diff
@@ -187,58 +183,57 @@ def _init_sae_group_b_decs(
     def save_checkpoint(
         self,
         trainer: SAETrainer,
-        checkpoint_name: int | str,
+        checkpoint_name: str,
         wandb_aliases: list[str] | None = None,
-    ) -> str:
-
-        checkpoint_path = f"{trainer.cfg.checkpoint_path}/{checkpoint_name}"
+    ) -> None:
+        """Save a checkpoint of the SAE locally and optionally to wandb."""
 
-        os.makedirs(checkpoint_path, exist_ok=True)
+        base_path = Path(self.cfg.checkpoint_path) / checkpoint_name
+        base_path.mkdir(exist_ok=True)
 
-        path = f"{checkpoint_path}"
-        os.makedirs(path, exist_ok=True)
+        self.activations_store.save_state(base_path)
 
+        # TODO(oli-clive-griffin): Is this broken? it seems like this is "assymetrical" in the sense that
+        # this changes the output of the SAE.
+        # Should we not also balance this by scaling the encoder weights?
         if self.sae.cfg.normalize_sae_decoder:
             self.sae.set_decoder_norm_to_unit_norm()
-        self.sae.save_model(path)
+
+        weights_path, cfg_path, sparsity_path = self.sae.save_model(
```

Comment:
by returning these paths, we're able to remove this module's knowledge of the internal mechanism of `self.sae.save_model` when reading the files it saves below. i.e. the below code doesn't need to know that the correct path is: `f"{path}/{SAE_WEIGHTS_PATH}"` in `model_artifact.add_file(f"{path}/{SAE_WEIGHTS_PATH}")`.